app/uk/gov/hmrc/bindingtariffrulingfrontend/utils/EnumJson.scala

```diff
@@ -17,10 +17,24 @@
 package uk.gov.hmrc.bindingtariffrulingfrontend.utils
 
 import play.api.libs.json._
+import scala.util.Try
 
 object EnumJson {
 
-  def format[E <: Enumeration](enumeratedValue: E): Format[E#Value] =
-    Format(Reads.enumNameReads(enumeratedValue), Writes.enumNameWrites)
+  private def enumReads[E <: Enumeration](`enum`: E): Reads[E#Value] = {
+    case JsString(s) =>
+      Try(JsSuccess(enum.withName(s))).recover {
+        case _: NoSuchElementException =>
+          JsError(
+            s"Expected an enumeration of type: '${enum.getClass.getSimpleName}', but it does not contain the name: '$s'"
+          )
+      }.get
+
+    case _ => JsError("String value is expected")
+  }
+
+  implicit def enumWrites[E <: Enumeration]: Writes[E#Value] = (v: E#Value) => JsString(v.toString)
+
+  implicit def format[E <: Enumeration](`enum`: E): Format[E#Value] = Format(enumReads(enum), enumWrites)
```

Comment:
Preferably write tests for this like was done for enums in tariff-classification-frontend